gradio/state_holder.py

```diff
@@ -19,6 +19,10 @@ def set_blocks(self, blocks: Blocks):
         self.blocks = blocks
         self.capacity = blocks.state_session_capacity
 
+    def reset(self, blocks: Blocks):
```

Comment:
Just wondering why the 2nd argument is needed? Why do you need to call `self.set_blocks(blocks)` again?

Reply:
The ids of the second Blocks will be different from the first one

Reply:
Got it